pkg/ccl/backupccl/restore_online.go

```diff
@@ -171,6 +172,22 @@ func sendAddRemoteSSTWorker(
 		}
 
 		for entry := range restoreSpanEntriesCh {
+			// If we're restoring a tenant, we need to move the end key of any spans
+			// that ended at the exclusive end of the tenant span, i.e. the start key
+			// of the next tenant ID, into the prefix of this tenant span instead so
+			// that that end key can be rewritten to the restoring tenant ID's prefix.
+			// We do this be replacing any key equal to TenantSpan.EndKey (tID+1) with
+			// tenantSpan.Key followed by keyMax, since keyMax sorts able all table
+			// keys, including within the subspace of a tenant span.
+			if entry.ElidedPrefix == execinfrapb.ElidePrefix_Tenant {
+				_, id, err := keys.DecodeTenantPrefix(entry.Span.Key)
+				if err != nil {
+					return err
+				}
+				if tSpan := keys.MakeTenantSpan(id); entry.Span.EndKey.Equal(tSpan.EndKey) {
+					entry.Span.EndKey = append(tSpan.Key, keys.MaxKey...)
+				}
+			}
```

Comment:
er, you have to do this transformation for each backup file entry too. I'll ad this logic to `rewriteSpan`.

Reply:
do we not assert those the way we do the entry spans?

Reply:
put another way: why is this green?

Reply:
we answered why this is green, which is that we restore the tenant to the same ID.